### ISSUE TITLE
issuse with batch fetching order books

### ISSUE THREAD

**Issue by johnxn** [Cheery22487/hive_arbitrage#3]
hi, in stead of geting buy/sell orders one token at a time, we could retrieve order books for all tokens in a single request, like this::
`            url = HE_NODE + "/contracts"
            json_params = {
                "jsonrpc": "2.0",
                "id": 42069,
                "method": "find",
                "params": {
                    "contract": "market",
                    "query": {
                        "symbol": {"$in": tokens}
                    },
                    "indexes": [{"index": "priceDec", "descending": False}], 
                    "limit": 1000,
                    "offset": 0,
                    "table": "sellBook"
                }
            }
`
This will save us a little more time, could it make our bot a little more profitable? 
Thanks again for sharing with your work, l learnt a lot from it.

**Comment by Cheery22487** [comment]
Yes, this is an excellent idea. I wasn’t aware it was possible to make this general API request.

Reducing the number of API calls would significantly increase the scan frequency. Additionally, decreasing the lag between retrieving the first sell orders and the last buy orders—which is present now—will reduce vulnerability to market movements.

I’m not certain how substantial the practical upside will be, but significantly reducing the number of API calls is certainly a big advantage on its own.

I’ll update it when I get around to it.

Thank you very much for the feedback!
